### PR TITLE
remove need for `is_type_of` or `resolve_type` with hardcoded typename field

### PR DESCRIPTION
**Pull request by emeryotopalik**

### CODE REVIEW

File: lib/absinthe/type/input_union.ex [hunk 1]
```diff
@@ -42,7 +31,7 @@ defmodule Absinthe.Type.InputUnion do
           name: binary,
           description: binary,
           types: [Type.identifier_t()],
-          resolve_type: (any, Absinthe.Resolution.t() -> atom | nil),
+          resolve_type: nil,
```
Comment (binaryseed) [inline lib/absinthe/type/input_union.ex]: might as well remove it from the `defstruct` below & from this `spec`

File: test/support/fixtures/arguments_schema.ex [hunk 2]
```diff
@@ -185,7 +182,17 @@ defmodule Absinthe.Fixtures.ArgumentsSchema do
           {:ok, "NESTED THAT #{thing}"}
 
         %{list_union: lists}, _ ->
-          {:ok, lists |> Enum.flat_map(&Map.values/1) |> Enum.join("&")}
+          # TODO FIX THIS
```
Comment (binaryseed) [inline test/support/fixtures/arguments_schema.ex]: this was just a quick way to test the list union, you could just update the test assertion

File: lib/absinthe/phase/schema.ex [hunk 3]
```diff
@@ -286,20 +286,22 @@ defmodule Absinthe.Phase.Schema do
...
-  defp evaluate_resolve_type(fields, resolve_type) do
-    fields
-    |> Enum.into(%{}, fn
-      %{name: name, input_value: %{normalized: %{value: value}}} -> {name, value}
-    end)
-    |> resolve_type.()
+  defp extract_typename(fields) do
+    input_fields =
+      fields
+      |> Enum.into(%{}, fn
```
Comment (binaryseed) [inline lib/absinthe/phase/schema.ex]: probably don't need to do this conversion into a map anymore, and just get the `typename` directly

**Review by binaryseed — COMMENTED** [review]
can also remove `is_type_of` from `input_object.ex`

File: test/support/fixtures/arguments_schema.ex (hunk 2 above)
Comment (emeryotopalik) [reply test/support/fixtures/arguments_schema.ex]: yah I know. I liked what it was though.

**Pull request merged**